upgrades/v5.0.4/README.md

```diff
@@ -0,0 +1,257 @@
+   make install
+```
+
+### Check the new neutron version, verify the latest commit hash
+```shell
+   $ neutrond version --long
+    commit: 4ae1215c197f5d00f2462f5ce48cf1eaf68bf938
+    cosmos_sdk_version: v0.50.11-neutron
+    go: go version go1.22.9 linux/amd64
+    name: neutron
+    version: 5.0.4
+   ...
+```
+
+### Or check checksum of the binary if you decided to [download it](https://github.com/neutron-org/neutron/releases/tag/v5.0.4)
+
+```shell
+$ shasum -a 256 neutrond-linux-amd64
+40663d2a113d639eedfc29670d0c4996c5143386dabfd559624b4a9679959a03  neutrond-linux-amd64
+```
+
+
+### Make sure you are using the proper version of libwasm
+
+You can check the version you are currently using by running the following command:
+```
+$ neutrond q wasm libwasmvm-version
+
+2.1.3
+```
+The proper version is `2.1.3`.
+
+**If the version on your machine is different you MUST change it immediately!**
+
+#### Ways to change libwasmvm
+
+- Use a statically built Neutrond binary from an official Neutron release: [https://github.com/neutron-org/neutron/releases/tag/v5.0.4](https://github.com/neutron-org/neutron/releases/tag/v5.0.4)
+- If you built Neutron binary by yourself, `libwasmvm` should be loaded dynamically in your binary and somehow, the wrong `libwasmvm` library was present on your machine. You can change it by downloading the proper one and linking it to the Neutron binary manually:
+1. download a proper version of `libwasmvm`:
+
+```
+$ wget https://github.com/CosmWasm/wasmvm/releases/download/v2.1.3/libwasmvm.x86_64.so
+```
+
+2. tell the linker where to find it:
+```
+$ export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/lib/
+```
+
+3. check that libwasmvm version is correct:
+```
+$ neutrond q wasm libwasmvm-version
+2.1.3
+```
+
+### Method I: Manual Upgrade
+
+Make sure Neutron v5.0.4 is installed by either downloading a [compatible binary](https://github.com/neutron-org/neutron/releases/tag/v5.0.4), or building from source. Building from source requires **Golang 1.22.x**.
+
+Run Neutron v5.0.2 till upgrade height, the node will panic:
+
+```shell
+ERR UPGRADE "v5.0.4" NEEDED at height: 18015000: upgrade to v5.0.4 and applying upgrade "v5.0.4" at height: 18015000
+```
+
+Stop the node, and switch the binary to **Neutron v5.0.4** and re-start by `neutrond start`.
+
+It may take several minutes to a few hours until validators with a total sum voting power > 2/3 to complete their node upgrades. After that, the chain can continue to produce blocks.
+
+### Method II: Upgrade using Cosmovisor
+
+### Manually preparing the binary
+
+##### Preparation
+
+Install the latest version of Cosmovisor (`1.5.0`):
+
+```shell
+go install cosmossdk.io/tools/cosmovisor/cmd/cosmovisor@v1.5.0
+```
+
+**Verify Cosmovisor Version**
+
+```shell
+cosmovisor version
+cosmovisor version: v1.5.0
+```
+
+Create a cosmovisor folder:
+
+create a Cosmovisor folder inside `$NEUTRON_HOME` and move Neutron v3.0.2 into `$NEUTRON_HOME/cosmovisor/genesis/bin`
```

Comment:
not sure what version is correct here but v3.0.2 doesn't seem to be the one, does it? v5.0.3 probably